src/QueryConditionGenerator.php

```diff
@@ -81,40 +105,79 @@ private function processGroup(ValuesGroup $group): array
         // FIXME Objects need type formatter. `elastic_search_value_transformer` (use extensions to configure types)
 
         foreach ($group->getFields() as $fieldName => $valuesBag) {
+            $propertyName = $this->mappings[$fieldName]->propertyName;
+            $field = $this->fieldSet->get($fieldName);
+
+            // TODO: hack, review and fix
+            $fieldType = $field->getType()->getInnerType();
+            $dateTransformer = new DateTimeToStringTransformer(null, 'UTC', 'Y-m-d');
+
             if ($valuesBag->hasSimpleValues()) {
-                $bool[$includingType][]['terms'] = [$this->mappings[$fieldName]->indexName => array_values($valuesBag->getSimpleValues())];
+                if ($fieldType instanceof DateType) {
+                    // date lookups behave like a range even for an exact value in Elasticsearch
+                    // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html#ranges-on-dates
+                    $values = array_values($valuesBag->getSimpleValues());
+                    foreach ($values as $idx => $value) {
+                        $valuesBag->removeSimpleValue($idx);
```

Comment:
Note a ValuesBag is not immutable, and the QueryConditionGenerator should not change the ValuesBag.
Better to clone this.

For the future: Usually when a Query generator needs something special is to use TypeExtensions with options like `elastic_search_structure_transformer` which receives the ValuesBag (or ValuesGroup, or both) and allows to change it's structure.

Reply:
I'm just working on this and it's annoying as it is, especially since it's only placeholder code.

Reply:
I did something similar for the Doctrine DBAL integration 😋 `query conversions`.
https://github.com/rollerworks/search-doctrine-dbal/blob/master/src/Query/QueryField.php#L96

For example the https://github.com/rollerworks/search-doctrine-dbal/blob/master/src/Extension/Conversion/AgeDateConversion.php allows to search with both an age and a date.